src/Compilers/Core/Portable/Emit/EditAndContinue/SymbolChanges.cs

```diff
@@ -375,11 +321,19 @@ SymbolChange fixChangeIfMemberIsReAdded(ITypeDefinitionMember item, SymbolChange
 
         protected abstract ISymbolInternal? GetISymbolInternalOrNull(ISymbol symbol);
 
+        public ISymbolInternal GetRequiredInternalSymbol(ISymbol? symbol)
+        {
+            Debug.Assert(symbol != null);
+            var result = GetISymbolInternalOrNull(symbol);
+            Debug.Assert(result != null);
```

Comment:
>Debug.Assert(result != null); [](http://example.com/codeflow?start=12&length=29)

Should this method throw an exception if `result != null` rather than asserting?